docs/PIPELINES-CONFIG.md

```diff
@@ -0,0 +1,269 @@
+
+### AI-RHDH-Installer: Manual Configuration
+
+#### Step 1: TektonConfig CR
+You should patch the `TektonConfig` CR called `config` with the extended configuration under [`tekton-config.json`](../resources/tekton-config.json)
+to setup Tekton for AI software template use and RHDH integration.
+
+Before patching you will need to provide an additional field `transparency.url` under `chain` to point to the rekor server as follows:
+```json
+{
+    "spec": {
+        ...
+        "chain": {
+            ...
+            "transparency.url": "http://rekor-server.<rhdh_namespace>.svc"
+        },
+        ...
+    }
+}
+```
+
+#### Step 2: RHDH kubernetes plugin service account
+
+As part of the `ai-rhdh-installer` a service account with a token secret was created in your desired namespace with the name `rhdh-kubernetes-plugin`, token secret should have a name pattern `rhdh-kubernetes-plugin-token-*`, keep note of this Secret.
+
+#### Step 3: Setting up deployment namespaces
+
+You will need to create the namespaces for the different app deployments. With a target namespace called `ai-rhdh`, these namespaces would be called `ai-rhdh-app-development`, `ai-rhdh-app-stage`, and `ai-rhdh-app-prod`. These namespaces can be created using the following:
+
+```sh
+APP_NAMESPACE=$NAMESPACE-app-<developer|stage|prod>
+cat <<EOF | kubectl apply -f - >/dev/null
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    argocd.argoproj.io/managed-by: $NAMESPACE
+  name: $APP_NAMESPACE
+EOF
+```
+
+#### Step 4: Setting up cosign secret under deployment namespaces
+
+First you will need to fetch the cosign public key from the `signing-secrets` secret in the `openshift-pipelines` namespace that was setup by the installer:
+
+```sh
+kubectl get secrets -n openshift-pipelines signing-secrets -o jsonpath='{.data.cosign\.pub}' 2>/dev/null
+```
+
+Take note of the public key, then use it to create the cosign secret under each of the deployment namespaces:
+
+```sh
+cat <<EOF | kubectl apply -f - >/dev/null
+apiVersion: v1
+data:
+    cosign.pub: <cosign_public_key>
+kind: Secret
+metadata:
+    labels:
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/part-of: tekton-chains
+        operator.tekton.dev/operand-name: tektoncd-chains
+    name: cosign-pub
+    namespace: $APP_NAMESPACE
+type: Opaque
+EOF
+```
+
+#### Step 5: Setting up gitops authentication secret under deployment namespaces
+
+If you are using GitLab, you will need to create the GitLab authentication secret with your GitLab PAT as follows:
+
+```sh
+kubectl -n $APP_NAMESPACE create secret generic "gitlab-auth-secret" \
+    --from-literal=password=<gitlab_pat> \
+    --from-literal=username=oauth2 \
+    --type=kubernetes.io/basic-auth \
+    --dry-run=client -o yaml | kubectl -n $APP_NAMESPACE apply --filename - --overwrite=true >/dev/null
+```
+
+Otherwise, you will need to create a Git authentication secret with your PAT (e.g. GitHub PAT) as follows:
```

Comment:
If I'm not wrong the PAT you'll need to apply here should have a specific list of permissions for the github org right? Or it's not the case here?

Reply:
> If I'm not wrong the PAT you'll need to apply here should have a specific list of permissions for the github org right? Or it's not the case here?

@thepetk Specific permissions in general, not towards the organization though the account with the PAT should have elevated permissions to the organization, I think just write maybe.

Reply:
> > If I'm not wrong the PAT you'll need to apply here should have a specific list of permissions for the github org right? Or it's not the case here?
> 
> @thepetk Specific permissions in general, not towards the organization though the account with the PAT should have elevated permissions to the organization, I think just write maybe.

RHDH org app creds will be used to administrate the org whereas the PAT is just to allow GitOps to perform actions against the existing resources if I'm understanding the setup correctly.

Reply:
> I think just write maybe.

Might be good to provide little context about the creds in case someone is trying to install and their token doesn't have the necessary perms?

Reply:
> > I think just write maybe.
> 
> Might be good to provide little context about the creds in case someone is trying to install and their token doesn't have the necessary perms?

@thepetk I'd prefer we link to a RHDH doc page which provide this information, e.g. 'More information about setting up the PAT token can be found <here|url.to.rhdh.docpage>', as the scope of these docs is to just provide instruction on how to use the installer and configure scripts.

Reply:
Yes that would be perfect!

Reply:
Done in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/7/commits/fa8bec73487eb86784456959ff288f94ff59a927